js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetLiveDataProvider.tsx

```diff
@@ -313,19 +313,29 @@ async function _batchedQueryAssets(
     };
   });
   onUpdatingOrUpdated();
-  const data = await _queryAssetKeys(client, assetKeys);
-  const fetchedTime = Date.now();
-  assetKeys.forEach((key) => {
-    lastFetchedOrRequested[tokenForAssetKey(key)] = {
-      fetched: fetchedTime,
-    };
-  });
-  setData(data);
-  onUpdatingOrUpdated();
-  isFetching = false;
-  const nextAssets = _determineAssetsToFetch();
-  if (nextAssets.length) {
-    _batchedQueryAssets(nextAssets, client, setData, onUpdatingOrUpdated);
+
+  function doNextFetch() {
+    isFetching = false;
+    const nextAssets = _determineAssetsToFetch();
+    if (nextAssets.length) {
+      _batchedQueryAssets(nextAssets, client, setData, onUpdatingOrUpdated);
+    }
+  }
+  try {
+    const data = await _queryAssetKeys(client, assetKeys);
+    const fetchedTime = Date.now();
+    assetKeys.forEach((key) => {
+      lastFetchedOrRequested[tokenForAssetKey(key)] = {
+        fetched: fetchedTime,
+      };
+    });
+    setData(data);
+    onUpdatingOrUpdated();
+    doNextFetch();
+  } catch (e) {
+    console.error(e);
+    // Retry fetching in 5 seconds if theres a network error
+    setTimeout(doNextFetch, 5000);
```

Comment:
I'm a little scared that this could crash /after/ the network call (eg: in setData) and land us in a loop fetching data over and over. Could we more tightly wrap this try/catch around just the case we want to retry (`_queryAssetKeys`)? There's so much that happens in lines 332-334 that I'm afraid this is retrying React errors and stuff like that too.

Reply:
I did think about this but (1) React updates won't happen in the same loop (they're deferred / batched in response to async events) and it should be safe to call this function even if it were to happen because the function would just no-op